projects/ADuCM3029_demo_ad5592r_ad5593r/src/aio_dio_pdmz.c

```diff
@@ -1575,3 +1639,134 @@ int32_t aiodio_process(struct aiodio_dev *dev)
+	float volt_val;
+
+	for(i = 0; i < 8; i++) {
+		itoa(i, (char *)buff, 10);
+		usr_uart_write_string(dev->board_cli->uart_device, (uint8_t*)"CH");
+		usr_uart_write_string(dev->board_cli->uart_device, buff);
+		usr_uart_write_string(dev->board_cli->uart_device, (uint8_t*)" ");
+	}
+	usr_uart_write_string(dev->board_cli->uart_device, (uint8_t*)"\n");
+
+	i = 0;
+	while(1) {
+		if(dev->board_device->channel_modes[i] != CH_MODE_ADC) {
+			usr_uart_write_string(dev->board_cli->uart_device,
+					      (uint8_t*)"N/A ");
+		} else {
+			dev->read_adc(dev->board_device, i, &adc_val);
+			adc_val &= 0xfff;
+			volt_val = lsb_val * adc_val;
+			aiodio_ftoa(buff, volt_val);
+			usr_uart_write_string(dev->board_cli->uart_device, buff);
+			usr_uart_write_string(dev->board_cli->uart_device, (uint8_t*)" ");
+		}
+		i++;
+		if(i >= 8) {
+			i = 0;
+			usr_uart_write_string(dev->board_cli->uart_device, (uint8_t*)"\n");
+			mdelay(500);
+		}
+		usr_uart_read_char(dev->board_cli->uart_device, &exit_char,
+				   &rdy, UART_NON_BLOCKING);
+		if((rdy) && (exit_char == 'q'))
+			return usr_uart_write_string(dev->board_cli->uart_device,
+						     (uint8_t*)"\nUser abort.\n");
+	}
+}
+
+/**
+ * Stream input data from an ADC input.
+ *
+ * If and input is not ADC it will exit with an error message.
+ *
+ * @param [in] dev     - The device structure.
+ * @param [in] chan_no - Channel number.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int32_t aiodio_analog_in_stream_chan(struct aiodio_dev *dev,
+		uint8_t chan_no)
+{
+	uint8_t exit_char, rdy;
+	uint8_t buff[20];
+	uint16_t adc_val;
+	float lsb_val = AD5592_3_INTERNAL_REFERENCE /
+			pow(2, AD5592_3_RESOLUTION_BITS);
+	float volt_val;
+
+	if(dev->board_device->channel_modes[chan_no] != CH_MODE_ADC)
+		return usr_uart_write_string(dev->board_cli->uart_device,
+					     (uint8_t*)"N/A ");
+
+	itoa(chan_no, (char *)buff, 10);
+	usr_uart_write_string(dev->board_cli->uart_device, (uint8_t*)"CH");
+	usr_uart_write_string(dev->board_cli->uart_device, buff);
+	usr_uart_write_string(dev->board_cli->uart_device, (uint8_t*)"\n");
+
+	while(1) {
+		dev->read_adc(dev->board_device, chan_no, &adc_val);
+		adc_val &= 0xfff;
+		volt_val = lsb_val * adc_val;
+		aiodio_ftoa(buff, volt_val);
+		usr_uart_write_string(dev->board_cli->uart_device, buff);
+		usr_uart_write_string(dev->board_cli->uart_device, (uint8_t*)"\n");
+
+		usr_uart_read_char(dev->board_cli->uart_device, &exit_char,
+				   &rdy, UART_NON_BLOCKING);
+		if((rdy) && (exit_char == 'q'))
+			return usr_uart_write_string(dev->board_cli->uart_device,
+						     (uint8_t*)"\nUser abort.\n");
+		mdelay(500);
+	}
+}
+
+/**
+ * Stream input data from the ADC inputs.
+ *
+ * @param [in] dev - The device structure.
+ * @param [in] arg - The number of the channel (0-7) all (a).
+ *
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t aiodio_analog_in_stream(struct aiodio_dev *dev, uint8_t *arg)
```

Comment:
You can use _char arg_ and it will be more suggestive for the user to use char data.

Reply:
This has to be a standard function signature to be passed as a user command to the command line interpreter. We have also made the choice to use bit specific types on the no-OS repositories. This is why I would argue to keep this as it is.